08-generic_cache/02_configured/cache.go

```diff
@@ -0,0 +1,48 @@
+package cache
+
+import "sync"
+
+// Cache is key, value storage.
+type Cache[K comparable, V any] struct {
+	data  map[K]V
+	mutex sync.Mutex
```

Comment:
Move the mutex above things that it protects. It's a nice convention to follow because it makes the lock intent clearer.